src/MAUI/Maui.Samples/Samples/Layers/AddVectorTiledLayerFromCustomStyle/AddVectorTiledLayerFromCustomStyle.xaml.cs

```diff
@@ -94,37 +95,31 @@ private async Task Initialize()
             }
         }
 
-        private void ButtonClick(object sender, EventArgs e)
+        private void StyleChooser_SelectedIndexChanged(object sender, EventArgs e)
         {
-            _ = StyleChooser();
+            // Get the style name and index of the selected item.
+            _ = ChangeStyle(StyleChooser.SelectedIndex, StyleChooser.SelectedItem.ToString());
         }
```

Comment:
I like this change!  The separation between app logic ("changing style") and GUI logic (interacting with specific controls) is better now.

Optional: I think ChangeStyle can be even cleaner if you moved try/catch/report out of it.  Something like...

```cs
private async void StyleChooser_SelectedIndexChanged(object sender, EventArgs e)
{
    try
    {
        // Get the style name and index of the selected item.
        await ChangeStyle(StyleChooser.SelectedIndex, StyleChooser.SelectedItem.ToString());
    }
    catch (Exception ex)
    {
        // Report exceptions.
        await Application.Current.MainPage.DisplayAlert("Error", ex.Message, "OK");
    }
}
```

This makes ChangeStyle into a simpler, more reusable method that only does what it says (changes style) without extra side-effects.